### PR TITLE
refactor: extract business logic

### PR DESCRIPTION
**Pull request by aradmargalit**
Our main page was getting pretty cluttered, this just pulls out functions to their own files for housekeeping

### CODE REVIEW

File: src/hooks/useInfiniteScroll.ts [hunk 1]
```diff
@@ -12,17 +12,18 @@ type FetchMoreOpts = {
  * fetchMore should fetch more results and update any state tied to your UI
  * returns how many items it fetched and if there is a next page
  */
-export type FetchMore = (_opts: FetchMoreOpts) => Promise<{
+export type FetchMore<T> = (_opts: FetchMoreOpts) => Promise<{
   itemsFetched: number;
   hasNextPage: boolean;
+  data: T | null;
```
Comment (aradmargalit) [inline src/hooks/useInfiniteScroll.ts:18]: Return the fetched data to allow consuming UI to react to these updates. Previously `fetchMore` itself needed to include the UI updater, but that made for unnecessarily tight coupling

File: src/hooks/useIsVisible.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Ref, useCallback, useEffect, useRef, useState } from 'react';
 
 export type UseIsVisibleResult<T extends Element> = {
-  ref: Ref<T | null>;
```
Comment (aradmargalit) [inline src/hooks/useIsVisible.ts:4]: This was never needed and allows us to remove a useless `@ts-ignore`

File: src/pages/p/strava-activities/index.tsx [hunk 3]
```diff
@@ -59,78 +59,29 @@ export const getServerSideProps: GetServerSideProps<{ data: Data }> = async ({
...
-        hasNextPage: false,
-      };
-    }
-  };
+  // Wraps fetchMore to commit the result to our UI
+  async function fetchMoreAndUpdate(opts: FetchMoreOpts) {
+    const result = await fetchMore(opts);
+    result.data && setActivities([...activities, ...result.data]);
+    return result;
+  }
```
Comment (aradmargalit) [inline src/pages/p/strava-activities/index.tsx:73]: I'm still not crazy about this, but at least now there's a clear boundary between "get more data" and "update the UI with new data".

**Review by eshedmargalit — APPROVED** [review]
LGTM, thanks!

**Pull request merged**